app/modules/intelligence/provider/provider_service.py

```
@@ -646,37 +646,46 @@ def _get_api_key(self, provider: str) -> str:
+                logger.debug(f"Secret Manager returned 404 for provider {provider}, trying fallback")
                 # Try provider-specific env var as fallback
                 env_key = os.getenv(f"{provider.upper()}_API_KEY")
                 if env_key:
+                    logger.info(f"Using {provider.upper()}_API_KEY environment variable for provider: {provider}")
                     self._api_key_cache[provider] = env_key
                     return env_key
                 # Cache None to indicate we've checked and it's not available
+                logger.warning(f"No API key found for provider {provider} after checking all sources")
                 self._api_key_cache[provider] = None
                 return None
+            logger.error(f"Error retrieving API key for provider {provider}: {str(e)}")
```

Ruff (0.14.14)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)